hub/core/storage/s3.py

```diff
@@ -2,34 +2,50 @@
 import botocore  # type: ignore
 import posixpath
 from typing import Optional
-from collections.abc import MutableMapping
+from hub.core.storage.provider import StorageProvider
 from hub.util.exceptions import S3GetError, S3SetError, S3DeletionError, S3ListError
 
 
-class S3Mapper(MutableMapping):
-    """An s3 mapper built using boto3. For internal use only (by class S3Provider)"""
+class S3Provider(StorageProvider):
+    """Provider class for using S3 storage."""
 
     def __init__(
         self,
-        url: str,
+        root: str,
         aws_access_key_id: Optional[str] = None,
         aws_secret_access_key: Optional[str] = None,
         aws_session_token: Optional[str] = None,
         endpoint_url: Optional[str] = None,
         aws_region: Optional[str] = None,
         max_pool_connections: Optional[int] = 10,
+        client=None,
     ):
+        """Initializes the S3Provider
+        Example:
```

Comment:
line breaks between sections please. this block of text is very hard to read at a glance